awx/ui_next/src/screens/Template/WorkflowJobTemplateDetail/WorkflowJobTemplateDetail.jsx

```diff
@@ -69,20 +67,22 @@ function WorkflowJobTemplateDetail({ template, i18n }) {
     </TextList>
   );
 
-  if (hasContentLoading) {
-    return <ContentLoading />;
-  }
-
-  const handleDelete = async () => {
-    setHasContentLoading(true);
-    try {
+  const {
+    request: deleteWorkflowJobTemplate,
+    isLoading,
+    error: deleteError,
+  } = useRequest(
+    useCallback(async () => {
       await WorkflowJobTemplatesAPI.destroy(id);
       history.push(`/templates`);
-    } catch (error) {
-      setDeletionError(error);
-    }
-    setHasContentLoading(false);
-  };
+    }, [id, history])
+  );
+
+  const { error, dismissError } = useDismissableError(deleteError);
+
+  if (isLoading) {
+    return <ContentLoading />;
+  }
```

Comment:
Same as JobTemplateDetail above: instead of rendering `<ContentLoading />`, keep the UI mounted & just disable the delete button while `isLoading` is true.

Reply:
Thanks for your review @keithjgrant, I will update per comments.